client/core/core.go

```diff
@@ -1647,6 +1676,8 @@ func (c *Core) ReconfigureWallet(appPW, newWalletPW []byte, assetID uint32, cfg
 	c.walletMtx.Unlock()
 
 	if oldWallet.connected() {
+		// NOTE: Cannot lock the wallet backend because it may be the same as
+		// the one just connected.
```

Comment:
Should really check if this is the same wallet and user has active trades. If both are yes error, can't use a different wallet with active trades.

We have the old wallet's deposit address, so maybe a simple `isMine` method on wallets would be good enough?

Also, after checking we will know if we should disconnect here or not.

Reply:
That's a great point about blocking the user from switching to a wallet that will break active orders or swaps! We should do that, but let's make an issue and another PR for it.  Scope is creeping on this PR.

Shutdown locking in the wrong place was causing consumers to hit a different bug in previous releases (the "not unlocked" one), and has encouraged the unfortunate habit of restarting wallets to avoid that now-fixed bug, which in turn caused another bug to manifest, which we also [just fixed](https://github.com/decred/dcrdex/pull/965).  😌 Phew.  Anyway, I want to close the book on this PromptShutdown design flaw as concisely as possible so it can backport into 0.1.5 with minimal trouble.